src/java.desktop/windows/native/libawt/windows/awt_Win32GraphicsDevice.cpp

```diff
@@ -179,7 +179,15 @@ void AwtWin32GraphicsDevice::Initialize()
     }
     gpBitmapInfo->bmiHeader.biBitCount = 0;
     HDC hBMDC = this->GetDC();
+    VERIFY(hBMDC != NULL);
+    if (hBMDC == NULL)
+        return;
     HBITMAP hBM = ::CreateCompatibleBitmap(hBMDC, 1, 1);
+    VERIFY(hBM != NULL);
+    if (hBM == NULL) {
+        VERIFY(::DeleteDC(hBMDC));
+        return;
+    }
     VERIFY(::GetDIBits(hBMDC, hBM, 0, 1, NULL, gpBitmapInfo, DIB_RGB_COLORS));
```

Comment:
Hm, I added some code for verification of ::GetDIBits result. If it is 0, I skip the rest of the coding. I don't want to touch more here since I don't understand what it is doing.

Reply:
It is not what I meant… If we want to preserve the way it work, we should execute the below code even if `this->GetDC()` fails…

On the other hand, testing didn't find any problems, in which case I prefer bailing out after `::GetDIBits` fails too.

@prrace, Do you know how critical the code below is? Is it worth to code the fallback if the path to `::GetDIBits`?

Reply:
> @prrace, Do you know how critical the code below is? Is it worth to code the fallback if the path to ::GetDIBits?

In the headLESS case we should never be here, since this is initializing the devices that don't exist in that case.
In the headFUL case I would prefer to keep the existing non-debug behaviour that we soldier on and try to do something sensible. Line 204 has a comment about a case in which GetDIBits really did fail and there's code to cope. I am writing specifically about GetDIBits.

Reply:
OK, I have reverted all changes to awt_Win32GraphicsDevice.cpp since we don't go through the code in the new headless case any more. I guess it's best to leave any refactoring of this code to later efforts.